packages/form-participation/src/FormControlMixin.ts

```diff
@@ -0,0 +1,517 @@
+     * For instance, if a given Validator has a `required` attribute, that
+     * validator will be evaluated whenever the host's required attribute
+     * is updated.
+     */
+    static get observedAttributes(): string[] {
+      const validatorAttributes = this.formControlValidators
+        .map(validator => validator.attribute);
+
+      /** @ts-ignore This exits */
+      const observedAttributes = super.observedAttributes || [];
+
+      /** Make sure there are no duplicates inside the attributes list */
+      const attributeSet = new Set([...observedAttributes, ...validatorAttributes])
+      return [...attributeSet];
+    }
+
+    /**
+     * Return the validator associated with a given attribute. If no
+     * Validator is associated with the attribute, it will return null.
+     */
+    static getValidator(attribute: string): Validator {
+      return this.formControlValidators.find(validator =>
+        validator.attribute === attribute
+      );
+    }
+
+    /** The ElementInternals instance for the control. */
+    internals = this.attachInternals() as unknown as IElementInternals;
+
+    /** Keep track of if the control has focus */
+    [focused] = false;
+
+    /**
+     * Exists to control when an error should be displayed
+     * @private
+     */
+     [forceError] = false;
+
+    /**
+     * Toggles to true whenever the element has been focused. This property
+     * will reset whenever the control's formResetCallback is called.
+     */
+    [touched] = false;
+
+    /**
+     * The element that will receive focus when the control's validity
+     * state is reported either by a form submission or via API
+     */
+    validationTarget: HTMLElement;
+
+    /**
+     * The controls' form value. As this property is updated, the form value
+     * will be updated. If a given control has a `checked` property, the value
+     * will only be set if `checked` is truthy.
+     */
+    value: any = '';
+
+    /** Return a reference to the control's form */
+    get form(): HTMLFormElement {
+      return this.internals.form;
+    }
+
+    /**
+     * Will return true if it is recommended that the control shows an internal
+     * error. If using this property, it is wise to listen for 'invalid' events
+     * on the element host and call preventDefault on the event. Doing this will
+     * prevent browsers from showing a validation popup.
+     */
+    get showError(): boolean {
+      return this[shouldShowError]();
+    }
+
+    /**
+     * The validation message shown by a given Validator object. If the control
+     * is in a valid state this should be falsy.
+     */
+    get validationMessage(): string {
+      return this.internals.validationMessage;
+    }
+
+    /**
+     * The element's validity state after evaluating the control's Validators.
+     * This property implements the same patterns as the built-in constraint
+     * validation strategy.
+     */
+    get validity(): ValidityState {
```

Comment:
Should/can we also cover the `checkValidity` method that we see on native `input` elements here?

Reply:
Adding.